### PR TITLE
INTDEV-529 Prevent card moving to under itself

### PR DESCRIPTION
**Pull request by Fuzzbender**
In data-handler, prevent card being moved to itself. 

Earlier, when this was done, the operation was successful and as last sub-operation 'source' is deleted, so this led to card being deleted. 

Additionally, prevent 'root' from being moved. Earlier it would complain that 'root' is not found from the project.

### CODE REVIEW

File: tools/data-handler/src/move.ts [hunk 1]
```diff
@@ -36,12 +39,18 @@ export class Move {
...
    * Moves card from 'destination' to 'source'.
    * @param source source card to move
-   * @param destination destination card where source card will be moved to; or 'root'
+   * @param destination destination card where source card will be moved to; or to root
    */
   public async moveCard(source: string, destination: string) {
+    if (source === ROOT) {
+      throw new Error('Cannot move "root"');
+    }
+    if (source === destination) {
```
Comment (samutoljamo-pareto) [inline tools/data-handler/src/move.ts:48]: Shouldn't we also check, that destination is not under the source card?

Comment (Fuzzbender) [reply tools/data-handler/src/move.ts:48]: > Shouldn't we also check, that destination is not under the source card?
Yes, I'll add that as a check as well.

Comment (Fuzzbender) [reply tools/data-handler/src/move.ts:48]: Fixed

File: tools/data-handler/src/move.ts [hunk 2]
```diff
@@ -61,6 +70,10 @@ export class Move {
       throw new Error(`Card ${destination} not found from project`);
     }
 
+    if (destinationCard.path.includes(source)) {
```
Comment (samutoljamo-pareto) [inline tools/data-handler/src/move.ts:73]: not in the scope of this, but: I think our path includes the full path on the system. Should we actually instead use path relative to project root? Having the whole system path might produce some weird bugs, if user has some folders, which have a meaning to us, higher up in the path

**Review by samutoljamo-pareto — APPROVED** [review]
Looks good to me.

File: tools/data-handler/src/move.ts (hunk 2 above)
Comment (Fuzzbender) [reply tools/data-handler/src/move.ts:73]: `Card` and `Resource` paths are indeed full paths in system. 
Both paths could be relative to the projectRoot.

Not going to change it here, but I'll make a JIRA ticket about it.

**Pull request merged**